sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -466,9 +468,9 @@ class Analyzer(
         // Find aggregate expressions and evaluate them early, since they can't be evaluated in a
         // Sort.
         val (withAggsRemoved, aliasedAggregateList) = newOrdering.map {
-          case aggOrdering if aggOrdering.collect { case a: AggregateExpression => a }.nonEmpty =>
+          case aggOrdering if aggOrdering.find(_.isInstanceOf[AggregateExpression]).isDefined =>
             val aliased = Alias(aggOrdering.child, "_aggOrdering")()
-            (aggOrdering.copy(child = aliased.toAttribute), Some(aliased))
+            (aggOrdering.copy(child = UnresolvedReference(aliased.exprId)), Some(aliased))
```

Comment:
cc @marmbrus , I created an `UnresolvedReference` to solve this issue.

In this case we need a "placeholder" here to reference to the still-need-resolve `Alias`. Previously we use `UnresolvedAttribute` as "placeholder" which depends on unique name(call `toAttribute` on an unresolved `Alias` will return an `UnresolvedAttribute` anyway), so I added a new kind of "placeholder" that depends on exprId.